students/kimeunji/westagram/users/views.py

```diff
@@ -1,3 +1,36 @@
-from django.shortcuts import render
+import json
+import re
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from users.models import User
+
+class UserView(View):
+    def post(self, request):
+        try:
+            data        = json.loads(request.body)
+            EMAIL_RE    = r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+            PASSWORD_RE = r'^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$'
+            email = data['email']
+            password = data['password']
+            
+            if not re.match(EMAIL_RE, email) :
+                return JsonResponse({'message':'INVALID_EMAIL'}, status = 400)
+        
+            if User.objects.filter(email = email).exists():
+                return JsonResponse({'message':'EMAIL_ALREADY_EXISTS'}, status = 400)
+                
+            if not re.match(PASSWORD_RE, password):
+                return JsonResponse({'message':'INVALID_PASSWORD'}, status = 400)
```

Comment:
```suggestion
            if not re.match(EMAIL_RE, email) :
                return JsonResponse({'message':'INVALID_EMAIL'}, status = 400)

            if not re.match(PASSWORD_RE, password):
                return JsonResponse({'message':'INVALID_PASSWORD'}, status = 400)

            if User.objects.filter(email = email).exists():
                return JsonResponse({'message':'EMAIL_ALREADY_EXISTS'}, status = 400)
```
정규식 적용이랑, 예외처리 무도 잘 하셨습니다.
한가지만 수정하면 더 좋을 것 같아요.

이메일 존재 여부 판단을 뒤로 옮겨주세요. 이메일 존재 여부는 DB와 통신을 해야 됩니다. 따라서 프론트로부터 받은 값들에 대한 검사를 모두 마친 뒤 통신이 필요한 검사들을 하는 것이 조금 더 효율적입니다.